### ISSUE TITLE
list()

### ISSUE THREAD

**Issue by ImprovementMajor** [ImprovementMajor/NewNew#3]
@bodrovis Hi Ilya! Please see the link below https://github.com/ImprovementMajor/NewNew/blob/master/ToDoTraining.py
I'm still wondering about list() function, since not sure what does it do in the homework from the above. I've searched the web, but it's still not clear. In our case do we use list() instead of keys() to get all the keys from the. dictionary?  I've tried to use it and not to use but the code worked anyway. And the purpose of this part  errand_list=To_do_list[date]  is to get all the значения но не ключи. But how does the program know how to connect значения с ключами in this part if date in To_do_list_keys:
    print("For this day you have the following errands")
    errand_list=To_do_list[date] <-------------- I don't understand how еhis code tells the program to get only значения 
    print(errand_list)
Thank you

**Comment by bodrovis** [comment]
When the `list` function is applied to a dictionary, you get a list with all the keys from this dictionary. This is just a trick that we're using for now because we haven't discussed the `.keys()` method key. For example:

```python
my_dict = {"a": 1, "b": 2}
keys = list(my_dict)
```

`keys` in the example above will contain `["a", "b"]`.

Actually, this is interesting because in that task we don't really require fetching the keys from the dictionary (though it is not a mistake). Why? Well, because the `in` operator works with dictionaries.

First example:

```python
my_dict = {"a": 1, "b": 2}
"a" in my_dict # => True
1 in my_dict # => False
```

What does that mean? It means that `in` checks whether a given key is present in the dictionary. Therefore `"a" in my_dict` this line returns `True` (there's such key). However, `in` does not work with values therefore `1 in my_dict` is `False` (`1` is a value not a key).

Returning to your program, you can say:

```python
To_do_list={
"2020-09-04": ["meet with friends", "go to the theater"],
"2020-02-01": ["get ready for the trip", "buy cat food"]
}
date=input("Please enter the date in format yyyy-mm-dd: ")
if date in To_do_list:
  # ... other code omitted
```

This is going to work right away because we are checking if the given date is used as a key.

Another example:

```python
my_dict = {"a": 1, "b": 2}
keys = list(my_dict)
"a" in keys # => True
1 in keys # => False
```

This works as well, because `keys` is the list with the dictionary keys, therefore the result is absolutely the same. Which means in your program you can also say:


```python
To_do_list={
"2020-09-04": ["meet with friends", "go to the theater"],
"2020-02-01": ["get ready for the trip", "buy cat food"]
}

to_do_keys = list(To_do_list)

date=input("Please enter the date in format yyyy-mm-dd: ")
if date in to_do_keys:
  # ... other code omitted
```

This is going to work as well. Effectively, those two examples simply show that the same task can be solved in different ways. Probably, the first solution is more compact.

Now, speaking about the `errand_list=To_do_list[date]` part. Take a look at the following example:

```python
To_do_list={
"2020-09-04": ["meet with friends", "go to the theater"],
"2020-02-01": ["get ready for the trip", "buy cat food"]
}
errands = To_do_list["2020-09-04"]
```

What the `errands` is going to contain? Obviously, it is going to contain a list `["meet with friends", "go to the theater"]`. This list is a value from the `To_do_list` dictionary, whereas the `"2020-09-04"` is the key. It means that the `errands = To_do_list["2020-09-04"]` line take the value from the dictionary.

Now we can take one more step:

```python
To_do_list={
"2020-09-04": ["meet with friends", "go to the theater"],
"2020-02-01": ["get ready for the trip", "buy cat food"]
}
date = "2020-09-04"
errands = To_do_list[date]
```

It works in the same way but they key is stored in the variable. However, the general idea is the same: you get a value by its key.

Finally:

```python
To_do_list={
"2020-09-04": ["meet with friends", "go to the theater"],
"2020-02-01": ["get ready for the trip", "buy cat food"]
}
date = input("Please enter the date in format yyyy-mm-dd: ")
errands = To_do_list[date]
```

If the user enters `"2020-09-04"`, the this code once again works in the same way and you will get the value under the `"2020-09-04"` key. Does it help? If no, then perhaps we can arrange a quick call to see this script in action.

As a sidenote: please do not name your variables with capital letters: `todo_list`, not `Todo_list` because capital case is used for classes.

**Comment by ImprovementMajor** [comment]
@bodrovis Thank you!

**Comment by ImprovementMajor** [comment]
@bodrovis Hello, Ilya! Why it gives me invalid syntax?
data=["apple", "lemon", "lime"]
for data in ["apple", "lemon", "lime"]
    print(data)

**Comment by bodrovis** [comment]
@ImprovementMajor Hi! I believe, you forgot `:`

```python
for data in ["apple", "lemon", "lime"]:
    print(data)
```

**Comment by ImprovementMajor** [comment]
@bodrovis Hi Ilya! Here is one more question from the previous homework. Please see arrow next to the code line. 
import random
cells={
"L":random.randint(1,0)
}
if cells ["L"]:   <-------------   we can always leave it open ended if it has random.randint? 
    cells["R"]:0
    cells["C"]:0
else:
    cells["C"]=random.randint(0,1)

    if cells["C"]:
        cells["R"]=0
    else:
        cells["R"]=1
guess=input("Please choose a cell:")
if guess in list(cells):
    if cells[guess]==1:
        print("You have guessed right")
    else:
        print("Try again")
else:
        print("Wrong cell")

**Comment by bodrovis** [comment]
Hello @ImprovementMajor :) Well, not quite. One thing, `random.randint(1,0)` should be `random.randint(0,1)` (because first you provide the lower border (0) and then the upper border (1). Of course, randint works with other number as well, so you could say `random.randint(5,150)` - it should generate a number between 5 and 150.

In our case the number will be 0 or 1. In that condition we can say `if cells["L"]` or we can say `if cells["L"] == 1` or `if cells["L"] != 0`. All three expressions would mean that same: the left cell has a ball in it, therefore other cells should be empty (because only one cell can contain a ball). Using either `if cells["L"] == 1` or `if cells["L"] != 0` is more explicit, so when in doubts I'd always recommend using more explicit code version.

But `if cells["L"]` will also work. This is because `cells["L"]` can only return 0 or 1 (because we are generating a random number between 0 and 1). So, if the number is `1`, we'll get `if 1` which is always True, because 1 is truthy. Effectively, 1 means "left cell has a ball in it".

To make sure, you can try running this code:

```python
number = 1
if number:
    print("this is true!")
else:
    print("this is false, but this line will never be executed!")
```

Okay, returning to our cell. If `cells["L"]` returns `0`, the condition effectively turns to just `if 0`. This condition is always False, because 0 is a falsey value (I believe, that was mentioned in the course a bit earlier). 0 in our example means "left cell has no ball in it". That's why we can use the shortcut version `cells["L"]`.

Basically, any number except for `0` and `0.0` is truthy: be it 1 or 100 or 999.

All lists are truthy as well, except for an empty list `[]`. Same goes for dictionaries and strings (these will be falsey only when they are empty).

To summarize, falsey values turn conditions to False. Truthy, of course, turn conditions to True.

**Comment by ImprovementMajor** [comment]
Thank you @bodrovisn , it's very clear now

**Comment by ImprovementMajor** [comment]
@bodrovis thank you!

**Comment by bodrovis** [comment]
Cool :)

**Comment by ImprovementMajor** [comment]
@bodrovis Hi Ilya! Could you please take a look at the additional questions?
https://github.com/ImprovementMajor/NewNew/commit/86b082659ab250c35f705532f8bde32bcc579bcb

**Comment by bodrovis** [comment]
https://github.com/ImprovementMajor/NewNew/issues/4 Let's discuss here, I'll check everything tomorrow
